### DISCUSSION TITLE
Redirects with wildcards

### DISCUSSION THREAD

**Discussion by devandnow** [vercel/next.js#36401] [Help] [answered]
Hi 

I have a number of redirects from an old php site where the incoming url has a dynamic pattern that I want to redirect.

/Bill-Jones
/Bill-Jones-Director
/Bill-Jones-Project-Wilson

There's an unknown number of options that are incoming after the /Bill-Jones that aren't on the new site

I want to make next redirect all urls starting with /Bill-Jones to /artists/bill-jones

What I can't work out is how to do a wildcard on the incoming url for the redirect.

I've tried this from the docs but to no avail

```
{
    source: '/Bill-Jones\\(*\\)',
    destination: '/artists/bill-jones',
    permanent: true,
}
```
All help appreciated

**Comment by vadymshymko** [comment, marked as answer]
Hi. Please, try the following code:

```
{
  source: '/Bill-Jones:path*',
  destination: '/artists/bill-jones',
  permanent: true,
},
```

**Reply by devandnow** [reply]
Perfect thank you!

**Comment by Arunjenson** [comment]
@vadymshymko consider this situation here i need the wildcard field for the destination
/demo/1
/demo/2
/demo/3

so it should to redirect like
for  /demo/1  => /example/1
 /demo/2 => /example/2
  /demo/3  => /example/3

**Reply by krystxf** [reply]
```ts
// next.config.ts

import type { NextConfig } from 'next'
 
const nextConfig: NextConfig = {
  async redirects() {
    return [ 
      // Wildcard path matching
      {
        source: '/blog/:slug',
        destination: '/news/:slug',
        permanent: true,
      },
    ]
  },
}
 
export default nextConfig
```
https://nextjs.org/docs/app/building-your-application/routing/redirecting#redirects-in-nextconfigjs
